Automattic/wp-calypso
# Remove unused v1 verticals code

#### Proposed Changes
Before team ganon's verticalisation project this year, a similar initiative was attempted but not released around ~2020. This "Verticals v1" code is no longer compatible with our current approach to v13n as it uses a different data model and list of verticals.

This removes traces of "Verticals v1" from calypso before removing it from wpcom and the API in 93-gh-Automattic/ganon-issues

see 93-gh-Automattic/ganon-issues

There ended up being a lot more legacy code than I realized making this quite a large PR. 

I removed the "Survey" signup step which didn't seem to be linked to any flow and has been around since 2015, using a hardcoded legacy verticals list.

I removed `verticals` from the `checklist` data-model. This was unused v1 vertical data coming from the `/sites/%s/checklist` endpoint used on my-home

There is still references to verticals in the following places:

in the theme selector, there seems to be a way to search by legacy vertical, but it doesn't seem to be made available by the UI.
in the ETK. apps/editing-toolkit/editing-toolkit-plugin/dotcom-fse/templates/class-wp-template-inserter.php I will  leave this for a follow up PR.

#### Testing Instructions
Test the unused `/start/design-first` flow which has had the "siteTopic" step removed.
Test the new `/start` flow and `/setup` flows to ensure nothing has been broken.
